system/jlib/jsocket.cpp

```diff
@@ -1175,8 +1186,12 @@ SocketEndpoint &CSocket::getPeerEndpoint(SocketEndpoint &ep)
             DBGLOG("getpeername failed %d",ERRNO());
             ep.set(NULL, 0);
         }
-        else
+        else {
+            CriticalBlock b(peerEpCrit);
             getSockAddrEndpoint(u,ul,ep);
```

Comment:
tirivial:  Unlikely to have any effect, but this could be before the critical section to reduce the window it is held for.

Reply:
I agree, move getSocketAddrEndpoint() above CB